### PR TITLE
[Merged by Bors] - Support LDAP authentication

### PR DESCRIPTION
**Pull request by maltesander**
# Description

- added LDAP authentication for Trino
- upgraded operator-rs

fixes https://github.com/stackabletech/trino-operator/issues/38

test: https://ci.stackable.tech/job/trino-operator-it-nightly/93/

Should https://github.com/stackabletech/trino-operator/issues/307 be done here as well?

## Review Checklist

- [x] Code contains useful comments
- [x] CRD change approved (or not applicable)
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/crd/src/authentication.rs [hunk 1]
```diff
@@ -98,26 +114,43 @@ impl TrinoAuthenticationMethod {
                     user_credentials: users,
                 })
             }
+            TrinoAuthenticationMethod::Ldap {
+                authentication_class: authentication_class_name,
+            } => {
+                let authentication_class =
+                    AuthenticationClass::resolve(client, authentication_class_name)
```
Comment (siegfriedweber) [inline rust/crd/src/authentication.rs:121]: The RBAC rules must be extended:

```
 - apiGroups:
     - authentication.stackable.tech
   resources:
     - authenticationclasses
   verbs:
     - get
     - list
     - watch
 ```

**Review by siegfriedweber — CHANGES_REQUESTED** [review]

File: rust/operator-binary/src/controller.rs [hunk 2]
```diff
@@ -162,6 +166,17 @@ pub enum Error {
         source: stackable_operator::error::Error,
         unit: String,
     },
+    #[snafu(display("illegal container name: [{container_name}]"))]
+    IllegalContainerName {
+        source: stackable_operator::error::Error,
+        container_name: String,
+    },
```
[the remark this answers is not in the document]
Comment (maltesander) [reply rust/operator-binary/src/controller.rs:173]: Yeah its static for now and it wont happen. Not sure if this ever changes but i dont like the operator to panic.

File: docs/modules/usage_guide/pages/security.adoc [hunk 3]
```diff
@@ -27,12 +53,85 @@ stringData:
...
+  workers:
+    roleGroups:
+      default:
+        replicas: 1
+----
+
+With this configuration, Trino will authenticate via `ldaps://openldap.default.svc.cluster.local:1636`.
+The UI or Trino CLI need to authenticate with the user `integrationtest` and password `integrationtest`.
+
+Please have a look at the LDAP https://github.com/stackabletech/nifi-operator/tree/main/tests/templates/kuttl/ldap[integration tests] for testing, setting up OpenLDAP and creating the integrationtest user.
```
[the remark this answers is not in the document]
Comment (maltesander) [reply docs/modules/usage_guide/pages/security.adoc:41]: Not sure, this is a complete working example, we could point to the docs but the user has to get familiar with the concepts if adaptations are needed?

File: docs/modules/usage_guide/pages/security.adoc [hunk 4]
```diff
@@ -1,65 +1,52 @@
...
-type: kubernetes.io/opaque
-stringData:
-  admin: $2y$10$89xReovvDLacVzRGpjOyAOONnayOgDAyIS2nW9bs5DJT98q17Dy5i
-  alice: $2y$10$HcCa4k9v2DRrD/g7e5vEz.Bk.1xg00YTEHOZjPX7oK3KqMSt2xT8W
-  bob: $2y$10$xVRXtYZnYuQu66SmruijPO8WHFM/UK5QPHTr.Nzf4JMcZSqt3W.2.
+include::example$code/trino-file-authentication.yaml[]
 ----
 
-The <user>:<password> combinations are provided in the `stringData` field. The hashes are created using bcrypt with 10 rounds.
+The `simple-trino-users-secret` points to a manually created secret. This contains username and password pairs as shown in the following snippet:
```
Comment (siegfriedweber) [inline docs/modules/usage_guide/pages/security.adoc:20]:
```suggestion
The `simple-trino-users-secret` points to a manually created secret. This contains username and password pairs as shown in the previous snippet.
```

**Review by siegfriedweber — CHANGES_REQUESTED** [review]

File: docs/modules/usage_guide/pages/security.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (maltesander) [reply docs/modules/usage_guide/pages/security.adoc:41]: Yeah i like that idea. And yeah it should have pointed to trino ... that was copy paste :D

[the remark this answers is not in the document]
Comment (siegfriedweber) [reply docs/modules/usage_guide/pages/security.adoc]: Ticket created: stackabletech/documentation#300

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**